Sources/BSON/BSONInt.swift

```diff
@@ -0,0 +1,35 @@
+import NIO
```

Comment:
I think for consistency it might be better if we named this file like `Integers+BSONValue.swift`.
and then you'll probably have similar files for like `Date+BSONValue.swift`, etc.
we often name protocol conformance files in that way.

Reply:
Yea that makes sense, I did breakout my ByteBuffer extensions into their own file `ByteBuffer+BSON.swift` but since I'm now renaming files to follow this format this would seem to imply it "implements" the BSON enum lol so I changed it to `ByteBuffer+SwiftBSON.swift` since there's no real conformance to list

I did find this tutorial that I can dig up again but basically it suggested when adding methods like this to an outside type you could instead confine your additional methods to a single property

```swift
class BSONHelper { func getCString {...} }
extension ByteBuffer {
  internal var asBSON: BSONHelper
}
``` 
Kinda like this example although there was a trick relating to extending it such that BSONHelper is the same type as ByteBuffer so that `self.` accessors still work. I couldn't get that to work when I was throwing it together to test, does that sound like something we'd like or even matters here? We could collab on the solution if so

Reply:
Yea I think its exactly that when the extensions are public, I agree we can keep the current way, and I'll rename 👍